wit/wasi-filesystem.wit.md

```diff
@@ -471,6 +471,11 @@ set-times: func(
 ```wit
 /// Read from a descriptor, without using and updating the descriptor's offset.
 ///
+/// This function returns a list of bytes containing the data that was
+/// read, along with a bool indicating whether the end of the file
+/// was reached. The returned list will contain up to `len` bytes; it
+/// may return fewer than requested, but not more.
```

Comment:
Perhaps clarify that it may short return, even when the end of the file is not reached? (since you mention the short reads right after the EOF, it could be interpreted that short read only occur when EOF is set).

Reply:
Yeah, that is pretty subtle. I've now added some more works to hopefully clarify that.